### ISSUE TITLE
Clarification on Kobo Support for Obfuscated Fonts

### ISSUE THREAD

**Issue by tooolbox** [kobolabs/epub-spec#17]
[Obfuscated Fonts Are Not Currently Supported by the Kobo CMS](https://github.com/kobolabs/epub-spec/blob/master/README.md#obfuscated-fonts-are-not-currently-supported-by-the-kobo-cms)
`````
Non-sideloaded titles with obfuscated fonts will display the reading
platform’s default font instead of the intended font. As a result they
may fail content QA. However, work is underway at Kobo to support
obfuscated fonts and render these titles correctly.
`````
I use InDesign CC to create an ePub file, although I do a lot of post-processing to set up the file for iBooks, Nook, Kindle, etc.  InDesign automatically applies the IDPF obfuscation system to all embedded fonts.  This font obfuscation method is handled just fine (at least when side-loading) to Nook, iBooks, etc.

When creating a Kindle file, KindleGen throws away obfuscated fonts, so I do have [a method of de-obfuscating](https://github.com/tooolbox/clarity) the fonts.  (They are later obfuscated/encrypted somewhere in the Amazon CMS, apparently.)  I can do the same "clarifying" process on the fonts, with the file that I submit to Kobo, if needed.

However, a number of the fonts that I am using are licensed from Adobe, and they have something of a requirement that any embedded fonts are properly obfuscated to prevent ripoff.  If I submit an ePub to the Kobo CMS that has unobfuscated fonts, are these obfuscated by the time they hit the readers?  The Kobo Touch seems to be able to handle such fonts in any case.

Thanks.

**Comment by bdugas** [comment]
Hello,
When ePubs are sent to Kobo and processed through its CMS the fonts themselves are not automatically obfuscated but the entire ePub is encrypted unless the publisher/account has chosen to distribute their book without DRM.
If the fonts you are licensing simply require that they not be distributed DRM free then you should have no issue. If the license requires that the fonts themselves be obfuscated on top of the DRM applied to the ePub then you could not use them for Kobo.

**Comment by tooolbox** [comment]
Great, that answers my question.  My only suggestion would be to clarify this in the documentation, but maybe it is obvious that, since the book has DRM, the fonts would be inaccessible under normal conditions.

Thanks!

**Comment by bdugas** [comment]
The section has been revised to clear up any ambiguity.

**Comment by tooolbox** [comment]
Fantastic, very clear now.  Thanks for your efforts!

**Comment by JayPanoz** [comment]
> If the license requires that the fonts themselves be obfuscated on top of the DRM applied to the ePub then you could not use them for Kobo.

OK, I don't want to be that guy but this issue is actually critical…

I know font licences are tough but the standard is “subset and obfuscate”. They don't even care about DRM.

Now, the vast majority of foundries use this standard. To the best of my knowledge, that’s also why InDesign is doing it by default when you export to EPUB.

Besides, in those font licences, you’ll almost always find some section telling “you can’t send a copy of the font to your client/subcontractor/distributor/etc.”

In other words, sending a file without obfuscated fonts to you is typically **a violation of the font licence.** What are we supposed to do? Typeset the fixed-layout eBook with libre fonts all over again? 

Please consider supporting obfuscated fonts ASAP, we’re currently stuck between a rock and a hard place.

**Comment by bdugas** [comment]
@JayPanoz we're well aware this puts you and all other producers of ePubs between a rock and a hard place when making FXL content for Kobo. What we've aimed to do with the wording in our guidelines and in our communications directly is to simply make it clear we don't support it and indicate how to get your content live on Kobo by swapping the fonts with non-obfuscated files or different fonts altogether.
That said we're well ahead of considering supporting font obfuscation. It's been one of our top items for increasing content support for some time but building support has required several backend and client updates. We have addressed our processing issues and now it's only our iOS app that can't handle obfuscated fonts in a processed ePub. We're aiming to have this fixed for a fall release and the sooner the better though we can't yet promise a date as it will be determined the many factors that always impact a release schedule.
